content/en/blog/2021/discovery-selectors/index.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Use DiscoverySelectors To Configure Namespaces For Your Istio Service Mesh"
+description: Learn how to use DiscoverySelectors and how it intersects with Sidecar resources.
+publishdate: 2021-04-30
+attribution: "Lin Sun (Solo.io), Christian Posta (Solo.io), Harvey Xia (Solo.io)"
+keywords: [discoveryselectors,Istio,namespaces,sidecar]
+---
+
+As users move their services to run in a service mesh, they are often surprised that the Istio control plane watches and processes all of the Istio and Kubernetes resources from all namespaces in the cluster by default. This can be an issue for very large clusters with lots of namespaces and deployments or even for a moderately sized cluster with rapidly churning resources (for example, Spark jobs).
+
+Both [in the community](https://github.com/istio/istio/issues/26679) as well as for our large-scale customers at [Solo.io](https://solo.io), we need a way to dynamically restrict the set of namespaces that are part of the mesh so that the Istio control plane only processes Istio and Kubernetes resources in those namespaces. The ability to restrict the namespaces enables Istiod to watch and push fewer resources and associated changes to the sidecars, thus improving the overall performance on the control plane and data plane.
+
+## Background
+
+By default, Istio watches all namespaces/services/endpoints/pods in a cluster. For example, in my Kubernetes cluster, I deployed the sleep service in the default namespace, and the httpbin service in the ns-x namespace. I’ve added the sleep service to the mesh, and I have no plan to add the httpbin service in the ns-x namespace to the mesh or have any service in the mesh interact with the httpbin service in the ns-x namespace.
+
+Use `istioctl pc endpoint` command to display all the endpoints for the sleep deployment:
+
+{{< image link="./endpoints-default.png" caption="Endpoints for Sleep Deployment" >}}
+
+Note that the httpbin service endpoint in the ns-x namespace is in the list of discovered endpoints. This may not be an issue when you only have a few services. However, when you have hundreds of services that don't interact with any of the services running in the Istio service mesh, you probably don't want your Istio control plane to watch these services and send their information to the sidecars of your services in the mesh.
```

Comment:
```suggestion
Note that the `httpbin` service endpoint in the `ns-x` namespace is in the list of discovered endpoints. This may not be an issue when you only have a few services. However, when you have hundreds of services that don't interact with any of the services running in the Istio service mesh, you probably don't want or need your control plane to watch these services, or send their information to the sidecars of your services in the mesh.
```